### DISCUSSION TITLE
Huge errors.ETIMEDOUT calls are appearing

### DISCUSSION THREAD

**Discussion by shiv-vsics** [artilleryio/artillery#2726] [Artillery]
Hi Team/ @hassy 

I am using Artillery for one of my API. I am hitting this API with load phases as 
duration: 300
ArrivalRate: 1000
![image](https://github.com/artilleryio/artillery/assets/66552369/a909a7aa-3d8f-4cb9-9c87-ad05084b9038)

but i can mostly all the calls are having errors.ETIMEDOUT. Could you please help me in knowing this why i am getting these much of TIMEDOUT request with above config.
I am running it on an Amazon EC2 Instance, m5 Large Instance. Attaching the snapshot for report as well.

![image](https://github.com/artilleryio/artillery/assets/66552369/8a63f6a1-eada-4bb2-b84d-2b65446964f2)

![image](https://github.com/artilleryio/artillery/assets/66552369/d0177e95-75f6-4739-8f8c-9924cb3c528e)

**Comment by hassy** [comment]
`ETIMEDOUT` means Artillery did not receive a response from the app under test quickly enough. This can happen for two reasons: (1) server oveload, and (2) client overload. Most of the time it's (1), but in your case it's probably (2) - `arrivalRate` of 1000 is probably too high. You're trying to create 1000 new VUs every second, which in turns creates 1000 new TCP connections every second and a bunch of other stuff, quite CPU heavy. You should try reducing `arrivalRate` and distribute your test instead - with AWS Lambda or AWS Fargate (both built into Artillery).
